spacy/lang/grc/punctuation.py

```diff
@@ -0,0 +1,52 @@
+from ..char_classes import LIST_PUNCT, LIST_ELLIPSES, LIST_QUOTES, LIST_CURRENCY
+from ..char_classes import LIST_ICONS, ALPHA_LOWER, ALPHA_UPPER, ALPHA, HYPHENS
+from ..char_classes import CONCAT_QUOTES
+
+_prefixes = (
+    [
+        "〈",
+        "†",
+        "⟦",
+        "⸏",
+    ]
+    + LIST_PUNCT
+    + LIST_ELLIPSES
+    + LIST_QUOTES
+    + LIST_CURRENCY
+    + LIST_ICONS
+)
+
+_suffixes = (
+    LIST_PUNCT
+    + LIST_ELLIPSES
+    + LIST_QUOTES
+    + LIST_ICONS
+    + [
+        "〉",
+        "†",
+        "⟧",
+        "⸎",
+        r"(?<=[\u1F00-\u1FFF-\u0370-\u03FF])\-",  # όνομα-
+        r"(?<=[\u1F00-\u1FFF-\u0370-\u03FF])\.",  # όνομα.
+        r"(?<=[\u1F00-\u1FFF-\u0370-\u03FF])⸏",  # ὄνομα⸏
```

Comment:
What exactly are you trying to achieve with these regular expressions? It might make sense to write a unit test to check the behaviour is correct (I can help implementing this if you want, but it'd be nice to understand your intention with some examples first).

Assuming you want to check for any unicode character in `\u1F00-\u1FFF` (Greek characters) and `\u0370-\u03FF` (Greek extended), we don't need the `-` between the two ranges. Further, we could simplify this expression by compressing into one line with `[\-\.⸏]` at the end, right?

Reply:
You are right here again: no need to use the - and this could be only one regex as you write. 

I added these lines because in several tests I ran the final punctuation was not always separated from the token. It was somehow inconsistent. The punctuation file for modern Greek has something similar. 

Example:

As right now with a model for grc trained with spacy 3.2, Σόλωνα⸏ is tokenized as Σόλωνα⸏ and Σόλωνα- as Σόλωνα-, while εἰς-Σόλωνα is tokenized as εἰς - Σόλωνα

In some random cases, Σόλωνα. was tokenized as Σόλωνα . but in others Σόλωνα. I do not know what is the reason for this inconsistency. 

I do not know how to write a test for this but I could provide the Greek examples.

Reply:
Regarding the regex [\-\.⸏] 

Maybe here ⸏ is redundant because it was already added above as a suffix.

Reply:
> I do not know how to write a test for this but I could provide the Greek examples.

That would be great. Basically if you can give me some short sentences or phrases where you encountered wrong tokenization, and then tell me how the ideal tokenization should look like, I can submit those as unit tests and we can make sure that the changes in this PR ensure that those examples work as you'd expect!

Reply:
Hi Sofie,

here are the sentences:

## Sentence 1

"τὸ 〈τῆς〉 φιλοσοφίας ἔργον ἔνιοί φασιν ἀπὸ ⟦βαρβάρων⟧ ἄρξαι."

**incorrect**: 

['Τὸ', '〈τῆς〉', 'φιλοσοφίας', 'ἔργον', 'ἔνιοί', 'φασιν', 'ἀπὸ', '⟦βαρβάρων⟧', 'ἄρξαι.']

**correct**:

['Τὸ', '〈', 'τῆς', '〉', 'φιλοσοφίας', 'ἔργον', 'ἔνιοί', 'φασιν', 'ἀπὸ', '⟦', 'βαρβάρων', '⟧', 'ἄρξαι', '.']

## Sentence 2

"τὴν δὲ τῶν Αἰγυπτίων φιλοσοφίαν εἶναι τοιαύτην περί τε †θεῶν† καὶ ὑπὲρ δικαιοσύνης."

**incorrect**:

['τὴν', 'δὲ', 'τῶν', 'Αἰγυπτίων', 'φιλοσοφίαν', 'εἶναι', 'τοιαύτην', 'περί', 'τε', '†θεῶν†', 'καὶ', 'ὑπὲρ', 'δικαιοσύνης.']

**correct**:

['τὴν', 'δὲ', 'τῶν', 'Αἰγυπτίων', 'φιλοσοφίαν', 'εἶναι', 'τοιαύτην', 'περί', 'τε', '†', 'θεῶν', '†', 'καὶ', 'ὑπὲρ', 'δικαιοσύνης', '.']

## Sentence 3

"⸏πόσις δ' Ἐρεχθεύς ἐστί μοι σεσωσμένος⸏"

**incorrect:**

['⸏πόσις', "δ'", 'Ἐρεχθεύς', 'ἐστί', 'μοι', 'σεσωσμένος⸏']

**correct:**

['⸏', 'πόσις', "δ'", 'Ἐρεχθεύς', 'ἐστί', 'μοι', 'σεσωσμένος', '⸏']



## Sentence 4

"⸏ὔπνον ἴδωμεν⸎"

**incorrect:**

['⸏ὔπνον', 'ἴδωμεν⸎']

**correct:**

['⸏', 'ὔπνον', 'ἴδωμεν', '⸎']

On my side I have adopted the single regex line that you suggested:

r"(?<=[\u1F00-\u1FFF\u0370-\u03FF])[\-\.⸏]",  # ὄνομα-